week-3-practise/level-3/1-practise-all.js

```diff
@@ -42,7 +42,55 @@ Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
 
 */
-
+var PreviousPassword = [
+  "fhD8!yrjj",
+  "ttkTu.wer3",
+  "dvyyeyY!5",
+  "qwbfj76%",
+  "tytT3729."
+];
 var password = ["Se%5", "TktE.TJTU", "384HsHF", "dvyyeyY!5", "tryT3729."];
 
-console.log(PasswordValidationResult)
\ No newline at end of file
+function controllLevell1(arr) {
```

Comment:
I tried running your solution but I was not able to. I'm posting myself solution so that you can compare.
```suggestion
function passwordValid1(password) {
    return (password.length > 4) 
    && password.match("[A-Z]") != null
    && password.match("[a-z]") != null
    && password.match("[0-9]") != null;
}

function passwordValid2(password) {
    return passwordValid1(password) && (password.match("[!#$%.]") != null);
}

function passwordValid3(password) {
    return passwordValid2(password) && !previousPassword.includes(password);
}

var PasswordValidationResult = password.map(passwordValid3);

console.log(PasswordValidationResult
```